### PR TITLE
[doc] Rework docs

### PR DESCRIPTION
**Pull request by maflcko**
- Minor formatting such as adjusting links
- Move sections of `doc/multiwallet-qt.md` to the source code and delete
  the file, as it is outdated
- Fix typo in the release notes
- Amend release process to mention update of BLOCK_CHAIN_SIZE

### CODE REVIEW

File: doc/README.md [hunk 1]
```diff
@@ -3,12 +3,14 @@ Bitcoin Core 0.13.99
...
 Setup
 ---------------------
-[Bitcoin Core](http://bitcoin.org/en/download) is the original Bitcoin client and it builds the backbone of the network. However, it downloads and stores the entire history of Bitcoin transactions (which is currently several GBs); depending on the speed of your computer and network connection, the synchronization process can take anywhere from a few hours to a day or more.
+Bitcoin Core is the original Bitcoin client and it builds the backbone of the network. However, it downloads and stores the entire history of Bitcoin transactions (which is currently several GBs); depending on the speed of your computer and network connection, the synchronization process can take anywhere from a few hours to a day or more.
 
 Running
 ---------------------
 The following are some helpful notes on how to run Bitcoin on your native platform.
 
+To download Bitcoin Core, visit [bitcoincore.org](https://bitcoincore.org/en/releases/).
```
Comment (laanwj) [inline doc/README.md]: I'd put this under "Setup" instead of "Running"

File: src/qt/bitcoingui.cpp [hunk 2]
```diff
@@ -74,6 +74,8 @@ const std::string BitcoinGUI::DEFAULT_UIPLATFORM =
 #endif
         ;
 
+/** Default wallet file name. Uses tilde to avoid name
```
Comment (laanwj) [inline src/qt/bitcoingui.cpp]: Calling this a file name may be deceptive. Maybe "display name for default wallet".

**Comment by laanwj** [comment]
ACK apart from nits.

**Comment by btcdrak** [comment]
ACK

**Comment by laanwj** [comment]
They've been fixed :)

**Pull request merged**